### PR TITLE
ISSUE-1048: Add Version Mixin

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
This PR adds version methods to the `OpenMetadata` class in `ometa_api.py` to support entity versioning.

### Type of change :
- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

Tests have not been added yet as we are waiting for [PR #1993](https://github.com/open-metadata/OpenMetadata/pull/1993)

### Reviewers
Ingestion: @harshach @ayush-shah @pmbrull

### CODE REVIEW

**Comment by harshach** [comment]
@TeddyCr the PR looks good to me. Are you planning on making any further changes as the PR in draft mode.

**Comment by TeddyCr** [comment]
> @TeddyCr the PR looks good to me. Are you planning on making any further changes as the PR in draft mode.

Hey @harshach thanks for following up on this PR. I was waiting for #1993 to be able to add integration tests to the PR 🙂. I just made it as ready for review.

File: ingestion/src/metadata/ingestion/ometa/mixins/versionMixin.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# pylint: disable=invalid-name
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/ometa/mixins/versionMixin.py:1]: We might need to update the naming of the classes 🤔

**Review by pmbrull — APPROVED** [review]
Thanks, @TeddyCr, looks great! 🚀 🎉

**Pull request merged**